src/systems/triggered_publisher/TriggeredPublisher.cc

```diff
@@ -0,0 +1,496 @@
+}
+//////////////////////////////////////////////////
+FieldMatcher::FieldMatcher(const std::string &_msgType, bool _logicType,
+                           const std::string &_fieldName,
+                           const std::string &_fieldString)
+    : matchMsg(msgs::Factory::New(_msgType)),
+      logicType(_logicType),
+      fieldName(_fieldName)
+{
+  if (nullptr == this->matchMsg)
+  {
+    return;
+  }
+
+  transport::ProtoMsg *matcherSubMsg{nullptr};
+  FindFieldSubMessage(this->matchMsg.get(), _fieldName, this->fieldDescMatcher,
+                      &matcherSubMsg);
+
+  if (this->fieldDescMatcher.empty())
+    return;
+  if (nullptr == matcherSubMsg)
+    return;
+
+  try
+  {
+    bool result = google::protobuf::TextFormat::ParseFieldValueFromString(
+        _fieldString, this->fieldDescMatcher.back(), matcherSubMsg);
+    if (!result)
+    {
+      ignerr << "Failed to parse matcher string ["
+             << _fieldString << "] for field [" << this->fieldName
+             << "] of input message type [" << _msgType << "]\n";
+      return;
+    }
+  }
+  catch (const std::exception &_err)
+  {
+    ignerr << "Creating Field matcher failed: " << _err.what() << std::endl;
+    return;
+  }
+
+  this->valid = true;
+}
+//////////////////////////////////////////////////
+bool FieldMatcher::FindFieldSubMessage(
+    transport::ProtoMsg *_msg, const std::string &_fieldName,
+    std::vector<const google::protobuf::FieldDescriptor *> &_fieldDesc,
+    transport::ProtoMsg **_subMsg)
+{
+  const google::protobuf::Descriptor *fieldMsgType = _msg->GetDescriptor();
+
+  // If fieldMsgType is nullptr, then this is not a composite message and we
+  // shouldn't be using a FieldMatcher
+  if (nullptr == fieldMsgType)
+    return false;
+
+  *_subMsg = _msg;
+
+  auto fieldNames = common::split(_fieldName, ".");
+  for (std::size_t i = 0; i < fieldNames.size(); ++i)
+  {
+    auto fieldDesc = fieldMsgType->FindFieldByName(fieldNames[i]);
+
+    if (nullptr == fieldDesc)
+      break;
+
+    _fieldDesc.push_back(fieldDesc);
+
+    if (google::protobuf::FieldDescriptor::CPPTYPE_MESSAGE !=
+        fieldDesc->cpp_type())
+    {
+      break;
+    }
+    fieldMsgType = fieldDesc->message_type();
+
+    if (nullptr == fieldMsgType)
+      break;
+
+    if (i < fieldNames.size() - 1)
+    {
+      auto *reflection = (*_subMsg)->GetReflection();
+      if (fieldDesc->is_repeated())
+      {
+        *_subMsg = reflection->AddMessage(*_subMsg, fieldDesc);
+      }
+      else
+      {
+        *_subMsg = reflection->MutableMessage(*_subMsg, fieldDesc);
+      }
+    }
+  }
+
+  return true;
+}
+
+//////////////////////////////////////////////////
+bool FieldMatcher::Match(
+    const transport::ProtoMsg &_input) const
+{
+  google::protobuf::util::DefaultFieldComparator comp;
+
+  auto *reflection = this->matchMsg->GetReflection();
+  const transport::ProtoMsg *subMsgMatcher = this->matchMsg.get();
+  const transport::ProtoMsg *subMsgInput = &_input;
```

Comment:
The `subMsgMatcher` should be valid if the matcher object is valid, which should be the case if this function is called. `subMsgInput` should be valid because its pointing to a parameter that was passed by reference.